ml/src/main/scala/frameless/ml/feature/TypedVectorAssembler.scala

```diff
@@ -0,0 +1,67 @@
+package frameless
+package ml
+package feature
+
+import org.apache.spark.ml.feature.VectorAssembler
+import org.apache.spark.ml.linalg.Vector
+import shapeless.{HList, HNil, LabelledGeneric}
+import shapeless.ops.hlist.ToTraversable
+import shapeless.ops.record.{Keys, Values}
+import shapeless._
+import scala.annotation.implicitNotFound
+
+final class TypedVectorAssembler[Inputs] private[ml](vectorAssembler: VectorAssembler, inputCols: Array[String])
+  extends AppendTransformer[Inputs, TypedVectorAssembler.Output, VectorAssembler] {
+
+  val transformer: VectorAssembler = vectorAssembler
+    .setInputCols(inputCols)
+    .setOutputCol(AppendTransformer.tempColumnName)
+
+}
+
+object TypedVectorAssembler {
+  case class Output(vector: Vector)
+
+  def create[Inputs]()(implicit inputsChecker: TypedVectorAssemblerInputsChecker[Inputs]): TypedVectorAssembler[Inputs] = {
+    new TypedVectorAssembler(new VectorAssembler(), inputsChecker.inputCols.toArray)
+  }
+}
+
+@implicitNotFound(
+  msg = "Cannot prove that ${Inputs} is a valid input type for TypedVectorAssembler. " +
+    "Input type must only contain numeric or boolean types."
+)
+private[ml] trait TypedVectorAssemblerInputsChecker[Inputs] {
+  val inputCols: Seq[String]
+}
+
+private[ml] object TypedVectorAssemblerInputsChecker {
+  implicit def checkInputs[Inputs, InputsRec <: HList, InputsKeys <: HList, InputsVals <: HList](
+    implicit
+    inputsGen: LabelledGeneric.Aux[Inputs, InputsRec],
+    inputsKeys: Keys.Aux[InputsRec, InputsKeys],
+    inputsKeysTraverse: ToTraversable.Aux[InputsKeys, Seq, Symbol],
+    inputsValues: Values.Aux[InputsRec, InputsVals],
+    inputsTypeCheck: TypedVectorAssemblerInputsValueChecker[InputsVals]
+  ): TypedVectorAssemblerInputsChecker[Inputs] = new TypedVectorAssemblerInputsChecker[Inputs] {
+    val inputCols: Seq[String] = inputsKeys.apply.to[Seq].map(_.name)
+  }
+}
+
+private[ml] trait TypedVectorAssemblerInputsValueChecker[InputsVals]
+
+private[ml] object TypedVectorAssemblerInputsValueChecker {
+  implicit def hnilCheckInputsValue: TypedVectorAssemblerInputsValueChecker[HNil] =
+    new TypedVectorAssemblerInputsValueChecker[HNil] {}
+
+  implicit def hlistCheckInputsValueNumeric[H, T <: HList](
+    implicit ch: CatalystNumeric[H],
+    tt: TypedVectorAssemblerInputsValueChecker[T]
+  ): TypedVectorAssemblerInputsValueChecker[H :: T] = new TypedVectorAssemblerInputsValueChecker[H :: T] {}
+
+  implicit def hlistCheckInputsValueBoolean[T <: HList](
```

Comment:
codecov say you this case is never in tests

Reply:
Mhh, it is tested in https://github.com/typelevel/frameless/pull/206/files#diff-8fe93d2237895b0ba271a37a5943580dR13 though. But I guess codecov can not go that far in implicit resolution. I'll try do write a test only with a Boolean in input.

Reply:
@atamborrino isn't there a typeclass already in shapeless for checking that all fields of type are subtypes of another type? This is related to TypedVectorAssemblerInputsValueChecker.